presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractFixedWidthBlock.java

```diff
@@ -171,7 +171,7 @@ public long getRegionSizeInBytes(int positionOffset, int length)
         if (positionOffset < 0 || length < 0 || positionOffset + length > positionCount) {
             throw new IndexOutOfBoundsException("Invalid position " + positionOffset + " in block with " + positionCount + " positions");
         }
-        return length * (fixedSize + Byte.BYTES);
+        return (long) length * (fixedSize + Byte.BYTES);
```

Comment:
It is not obvious whether this is correct: `(long) x * y`. I would rewrite it to either `y * (long) x` or `x * (long) y`.